tools/ea/Ea.hs

```diff
@@ -101,85 +106,59 @@ main = void $ do
       , mainnet7
       , mainnet8
       , mainnet9
+      , mainnetKAD
       ]
 
-show_ :: GChainId -> String
-show_ = \case
-    N -> "all chains"
-    KAD -> "chains 10-19"
-    n -> "Chain " <> show n
+show_ :: ChainIdRange -> String
+show_ (ChainIdRange n n')
+    | n == n' = "Chain " <> show n
+    | otherwise = "Chains " <> show n <> "-" <> show n'
+
+fullGenesisTag :: Genesis -> Text
+fullGenesisTag (Genesis _ tag cidr _ _ _ _ _) = tag <> T.pack (chainIdRangeTag cidr)
 
--- | Generate paylaods for a traversable of txs
+-- | Generate payloads for a traversable of txs
 --
 mkPayloads :: Traversable t => t Genesis -> IO ()
-mkPayloads = traverse_ mkPayload
+mkPayloads = traverse_ (\g -> writePayload g =<< mkPayload g)
 
--- | Generate a payload for a given genesis transaction
+writePayload :: Genesis -> Text -> IO ()
+writePayload gen payload = do
+    let fileName = "src/Chainweb/BlockHeader/Genesis/" <> fullGenesisTag gen <> "Payload.hs"
+    TIO.writeFile (T.unpack fileName) payload
+
+-- | Generate a payload for a given list of genesis transactions
 --
-mkPayload :: Genesis -> IO ()
-mkPayload (Genesis v tag cid c k a ns) = do
-    printf ("Generating Genesis Payload for %s on " <> show_ cid <> "...\n") $ show v
-    genPayloadModule v (tag <> sshow cid) txs
+mkPayload :: Genesis -> IO Text
+mkPayload gen@(Genesis v _ cidr@(ChainIdRange l u) c k a ns cc) = do
+    printf ("Generating Genesis Payload for %s on " <> show_ cidr <> "...\n") $ show v
+    payloadModules <- for [l..u] $ \cid ->
+        genPayloadModule v (fullGenesisTag gen) (unsafeChainId cid) =<< mkChainwebTxs txs
+    evaluate $ the payloadModules
```

Comment:
This is a check that, if we want the genesis payload to work on multiple chains, it must have the same result on all of those chains.